keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signers/hashicorp/dsl/HashicorpLocalNode.java

```diff
@@ -0,0 +1,276 @@
+    unseal(hashicorpVaultTokens.getUnsealKey());
+    login(hashicorpVaultTokens.getRootToken());
+    enableHashicorpKeyValueV2Engine();
+  }
+
+  // Start vault server. Sets port variable once started.
+  private void startVaultServer() throws IOException, CertificateEncodingException {
+    LOG.info("Starting vault server");
+    final AtomicBoolean serverStarted = new AtomicBoolean(false);
+    serverFuture =
+        new ProcessExecutor(vaultBinary.toString(), "server", "-config=" + getVaultServerConfig())
+            .readOutput(true)
+            .redirectOutput(
+                new LogOutputStream() {
+                  @Override
+                  protected void processLine(final String line) {
+                    LOG.info(line);
+
+                    // process Vault port (i.e. cluster port - 1)
+                    if (line.contains("cluster address")) {
+                      final Matcher matcher = portPattern.matcher(line);
+                      matcher.find();
+                      port.set(Integer.parseInt(matcher.group(1)) - 1);
+                    }
+
+                    if (line.contains("Vault server started!")) {
+                      serverStarted.set(true);
+                    }
+                  }
+                })
+            .start()
+            .getFuture();
+
+    Awaitility.await().atMost(5, TimeUnit.SECONDS).untilTrue(serverStarted);
+  }
+
+  @Override
+  public void shutdown() {
+    if (serverFuture != null) {
+      LOG.info("Shutting down Vault server Process...");
+      serverFuture.cancel(true);
+      serverFuture = null;
+    } else {
+      LOG.info("Vault server process is not running.");
+    }
+  }
+
+  @Override
+  public String getVaultToken() {
+    return vaultToken;
+  }
+
+  @Override
+  public String getHost() {
+    return host;
+  }
+
+  @Override
+  public int getPort() {
+    return port.intValue();
+  }
+
+  @Override
+  public Optional<SelfSignedCertificate> getServerCertificate() {
+    return serverTlsCertificate;
+  }
+
+  @Override
+  public void addSecretsToVault(final Map<String, String> entries, final String path) {
+    LOG.info("creating the secret in vault that contains the private key.");
+    final String secretPutPath = String.join("/", VAULT_ROOT_PATH, path);
+    for (final Map.Entry<String, String> entry : entries.entrySet()) {
+      try {
+        final String commandOutput =
+            executeCommandAndGetOutput(
+                hashicorpVaultCommands.putSecretCommand(
+                    entry.getKey(), entry.getValue(), secretPutPath));
+
+        assertThat(commandOutput.contains("created_time")).isTrue();
+      } catch (final IOException | InterruptedException | TimeoutException e) {
+        throw new RuntimeException("Error adding secrets to vault", e);
+      }
+
+      LOG.info("The secret ({}) was created successfully.", entry.getKey());
+    }
+  }
+
+  @Override
+  public String getHttpApiPathForSecret(final String secretPath) {
```

Comment:
this _should_ be common to both types of Hashicorp node

Reply:
moved to interface.